test/integration/default/controls/_mapdata_spec.rb

```diff
@@ -0,0 +1,25 @@
+# frozen_string_literal: true
+
+# Keep only first 2 digits from Ubuntu finger
+mapdata_file = "_mapdata/#{system.platform[:finger].split('.').first}.yaml"
+
+# Load the mapdata from profile https://docs.chef.io/inspec/profiles/#profile-files
+mapdata_dump = inspec.profile.file(mapdata_file)
+
+# Derive the location of the dumped mapdata
+output_dir =
+  case platform[:family]
+  when 'windows'
+    '/temp'
+  else
+    '/tmp'
+  end
+output_file = "#{output_dir}/salt_mapdata_dump.yaml"
+
+control '`map.jinja` YAML dump' do
+  title 'should contain the lines'
+
+  describe file(output_file).content.encode(universal_newline: true) do
```

Comment:
As a matter of style to limit the line size, we could do something like:

```ruby
output_file = "#{output_dir}/salt_mapdata_dump.yaml"
output_content = file(output_file).content.encode(universal_newline: true)

control '`map.jinja` YAML dump' do
  title 'should contain the lines'

  describe output_content do
    it { should eq mapdata_dump }
  end
end
```

Reply:
@baby-gnu This did cross my mind but I avoided it for the time being.  I _generally_ avoid single-use variables wherever possible -- I'm fine either way in this situation.

The other issue that distracted me is that the diff no longer seems to be showing, when the test fails.  I'm not sure if it's because of this new method of using `.content.encode(universal_newline: true)`, whereas I don't think the diff stopped working when I was using the original `mapdata_dump.gsub("\n", "\r\n")` method.

Reply:
Maybe we should use `.encode(universal_newline: true)` on `mapdata_dump` too?

Reply:
Investigating this diff issue and actually both methods work -- it's actually the latest (InSpec-related) gems that have stopped providing diffs.  So I tested with the bundle at [v0.16.3](https://github.com/saltstack-formulas/openvpn-formula/tree/v0.16.3) and the diff is working fine for both methods.  So the problem is somewhere here:

```diff
--------------------------------- Gemfile.lock ---------------------------------
index 23ed704..2adde2e 100644
@@ -1,6 +1,14 @@
+GIT
+  remote: https://github.com/test-kitchen/kitchen-docker.git
+  revision: 41e80fed3a7cc86323e19c16a5a340cebf7e5848
+  ref: 41e80fe
+  specs:
+    kitchen-docker (2.10.0)
+      test-kitchen (>= 1.0.0)
+
 GEM
   remote: https://rubygems.org/
   specs:
-    activesupport (5.2.4.3)
+    activesupport (5.2.4.4)
       concurrent-ruby (~> 1.0, >= 1.0.2)
       i18n (>= 0.7, < 2)
@@ -10,181 +18,187 @@ GEM
       public_suffix (>= 2.0.2, < 5.0)
     aws-eventstream (1.1.0)
-    aws-partitions (1.326.0)
-    aws-sdk-apigateway (1.42.0)
-      aws-sdk-core (~> 3, >= 3.71.0)
+    aws-partitions (1.383.0)
+    aws-sdk-apigateway (1.55.0)
+      aws-sdk-core (~> 3, >= 3.109.0)
       aws-sigv4 (~> 1.1)
-    aws-sdk-apigatewayv2 (1.21.0)
-      aws-sdk-core (~> 3, >= 3.71.0)
+    aws-sdk-apigatewayv2 (1.29.0)
+      aws-sdk-core (~> 3, >= 3.109.0)
       aws-sigv4 (~> 1.1)
-    aws-sdk-athena (1.27.0)
-      aws-sdk-core (~> 3, >= 3.71.0)
+    aws-sdk-athena (1.33.0)
+      aws-sdk-core (~> 3, >= 3.109.0)
       aws-sigv4 (~> 1.1)
     aws-sdk-autoscaling (1.22.0)
       aws-sdk-core (~> 3, >= 3.52.1)
       aws-sigv4 (~> 1.1)
-    aws-sdk-budgets (1.30.0)
-      aws-sdk-core (~> 3, >= 3.71.0)
+    aws-sdk-budgets (1.36.0)
+      aws-sdk-core (~> 3, >= 3.109.0)
       aws-sigv4 (~> 1.1)
-    aws-sdk-cloudformation (1.36.0)
-      aws-sdk-core (~> 3, >= 3.71.0)
+    aws-sdk-cloudformation (1.44.0)
+      aws-sdk-core (~> 3, >= 3.109.0)
       aws-sigv4 (~> 1.1)
-    aws-sdk-cloudhsm (1.22.0)
-      aws-sdk-core (~> 3, >= 3.71.0)
+    aws-sdk-cloudfront (1.44.0)
+      aws-sdk-core (~> 3, >= 3.109.0)
       aws-sigv4 (~> 1.1)
-    aws-sdk-cloudhsmv2 (1.23.0)
-      aws-sdk-core (~> 3, >= 3.71.0)
+    aws-sdk-cloudhsm (1.27.0)
+      aws-sdk-core (~> 3, >= 3.109.0)
       aws-sigv4 (~> 1.1)
-    aws-sdk-cloudtrail (1.23.0)
-      aws-sdk-core (~> 3, >= 3.71.0)
+    aws-sdk-cloudhsmv2 (1.30.0)
+      aws-sdk-core (~> 3, >= 3.109.0)
       aws-sigv4 (~> 1.1)
-    aws-sdk-cloudwatch (1.38.0)
-      aws-sdk-core (~> 3, >= 3.71.0)
+    aws-sdk-cloudtrail (1.29.0)
+      aws-sdk-core (~> 3, >= 3.109.0)
       aws-sigv4 (~> 1.1)
-    aws-sdk-cloudwatchlogs (1.31.0)
-      aws-sdk-core (~> 3, >= 3.71.0)
+    aws-sdk-cloudwatch (1.45.0)
+      aws-sdk-core (~> 3, >= 3.109.0)
       aws-sigv4 (~> 1.1)
-    aws-sdk-codecommit (1.33.0)
-      aws-sdk-core (~> 3, >= 3.71.0)
+    aws-sdk-cloudwatchlogs (1.38.0)
+      aws-sdk-core (~> 3, >= 3.109.0)
       aws-sigv4 (~> 1.1)
-    aws-sdk-codedeploy (1.31.0)
-      aws-sdk-core (~> 3, >= 3.71.0)
+    aws-sdk-codecommit (1.40.0)
+      aws-sdk-core (~> 3, >= 3.109.0)
       aws-sigv4 (~> 1.1)
-    aws-sdk-codepipeline (1.31.0)
-      aws-sdk-core (~> 3, >= 3.71.0)
+    aws-sdk-codedeploy (1.37.0)
+      aws-sdk-core (~> 3, >= 3.109.0)
       aws-sigv4 (~> 1.1)
-    aws-sdk-configservice (1.45.0)
-      aws-sdk-core (~> 3, >= 3.71.0)
+    aws-sdk-codepipeline (1.37.0)
+      aws-sdk-core (~> 3, >= 3.109.0)
       aws-sigv4 (~> 1.1)
-    aws-sdk-core (3.98.0)
+    aws-sdk-configservice (1.53.0)
+      aws-sdk-core (~> 3, >= 3.109.0)
+      aws-sigv4 (~> 1.1)
+    aws-sdk-core (3.109.1)
       aws-eventstream (~> 1, >= 1.0.2)
       aws-partitions (~> 1, >= 1.239.0)
       aws-sigv4 (~> 1.1)
       jmespath (~> 1.0)
-    aws-sdk-costandusagereportservice (1.21.0)
-      aws-sdk-core (~> 3, >= 3.71.0)
+    aws-sdk-costandusagereportservice (1.28.0)
+      aws-sdk-core (~> 3, >= 3.109.0)
+      aws-sigv4 (~> 1.1)
+    aws-sdk-dynamodb (1.55.0)
+      aws-sdk-core (~> 3, >= 3.109.0)
       aws-sigv4 (~> 1.1)
-    aws-sdk-dynamodb (1.48.0)
-      aws-sdk-core (~> 3, >= 3.71.0)
+    aws-sdk-ec2 (1.200.0)
+      aws-sdk-core (~> 3, >= 3.109.0)
       aws-sigv4 (~> 1.1)
-    aws-sdk-ec2 (1.165.0)
-      aws-sdk-core (~> 3, >= 3.71.0)
+    aws-sdk-ecr (1.39.0)
+      aws-sdk-core (~> 3, >= 3.109.0)
       aws-sigv4 (~> 1.1)
-    aws-sdk-ecr (1.30.0)
-      aws-sdk-core (~> 3, >= 3.71.0)
+    aws-sdk-ecs (1.70.0)
+      aws-sdk-core (~> 3, >= 3.109.0)
       aws-sigv4 (~> 1.1)
-    aws-sdk-ecs (1.63.0)
-      aws-sdk-core (~> 3, >= 3.71.0)
+    aws-sdk-efs (1.36.0)
+      aws-sdk-core (~> 3, >= 3.109.0)
       aws-sigv4 (~> 1.1)
-    aws-sdk-efs (1.29.0)
-      aws-sdk-core (~> 3, >= 3.71.0)
+    aws-sdk-eks (1.45.0)
+      aws-sdk-core (~> 3, >= 3.109.0)
       aws-sigv4 (~> 1.1)
-    aws-sdk-eks (1.37.0)
-      aws-sdk-core (~> 3, >= 3.71.0)
+    aws-sdk-elasticache (1.44.0)
+      aws-sdk-core (~> 3, >= 3.109.0)
       aws-sigv4 (~> 1.1)
-    aws-sdk-elasticache (1.36.0)
-      aws-sdk-core (~> 3, >= 3.71.0)
+    aws-sdk-elasticbeanstalk (1.38.0)
+      aws-sdk-core (~> 3, >= 3.109.0)
       aws-sigv4 (~> 1.1)
-    aws-sdk-elasticbeanstalk (1.31.0)
-      aws-sdk-core (~> 3, >= 3.71.0)
+    aws-sdk-elasticloadbalancing (1.29.0)
+      aws-sdk-core (~> 3, >= 3.109.0)
       aws-sigv4 (~> 1.1)
-    aws-sdk-elasticloadbalancing (1.22.0)
-      aws-sdk-core (~> 3, >= 3.71.0)
+    aws-sdk-elasticloadbalancingv2 (1.53.0)
+      aws-sdk-core (~> 3, >= 3.109.0)
       aws-sigv4 (~> 1.1)
-    aws-sdk-elasticloadbalancingv2 (1.44.0)
-      aws-sdk-core (~> 3, >= 3.71.0)
+    aws-sdk-elasticsearchservice (1.43.0)
+      aws-sdk-core (~> 3, >= 3.109.0)
       aws-sigv4 (~> 1.1)
-    aws-sdk-elasticsearchservice (1.36.0)
-      aws-sdk-core (~> 3, >= 3.71.0)
+    aws-sdk-firehose (1.35.0)
+      aws-sdk-core (~> 3, >= 3.109.0)
       aws-sigv4 (~> 1.1)
-    aws-sdk-firehose (1.28.0)
-      aws-sdk-core (~> 3, >= 3.71.0)
+    aws-sdk-guardduty (1.42.0)
+      aws-sdk-core (~> 3, >= 3.109.0)
       aws-sigv4 (~> 1.1)
-    aws-sdk-iam (1.39.0)
-      aws-sdk-core (~> 3, >= 3.71.0)
+    aws-sdk-iam (1.46.0)
+      aws-sdk-core (~> 3, >= 3.109.0)
       aws-sigv4 (~> 1.1)
-    aws-sdk-kafka (1.21.0)
-      aws-sdk-core (~> 3, >= 3.71.0)
+    aws-sdk-kafka (1.29.0)
+      aws-sdk-core (~> 3, >= 3.109.0)
       aws-sigv4 (~> 1.1)
-    aws-sdk-kinesis (1.23.0)
-      aws-sdk-core (~> 3, >= 3.71.0)
+    aws-sdk-kinesis (1.30.0)
+      aws-sdk-core (~> 3, >= 3.109.0)
       aws-sigv4 (~> 1.1)
-    aws-sdk-kms (1.33.0)
-      aws-sdk-core (~> 3, >= 3.71.0)
+    aws-sdk-kms (1.39.0)
+      aws-sdk-core (~> 3, >= 3.109.0)
       aws-sigv4 (~> 1.1)
-    aws-sdk-lambda (1.42.0)
-      aws-sdk-core (~> 3, >= 3.71.0)
+    aws-sdk-lambda (1.51.0)
+      aws-sdk-core (~> 3, >= 3.109.0)
       aws-sigv4 (~> 1.1)
     aws-sdk-organizations (1.17.0)
       aws-sdk-core (~> 3, >= 3.39.0)
       aws-sigv4 (~> 1.0)
-    aws-sdk-rds (1.85.0)
-      aws-sdk-core (~> 3, >= 3.71.0)
+    aws-sdk-rds (1.104.0)
+      aws-sdk-core (~> 3, >= 3.109.0)
       aws-sigv4 (~> 1.1)
-    aws-sdk-redshift (1.43.0)
-      aws-sdk-core (~> 3, >= 3.71.0)
+    aws-sdk-redshift (1.50.0)
+      aws-sdk-core (~> 3, >= 3.109.0)
       aws-sigv4 (~> 1.1)
-    aws-sdk-route53 (1.35.0)
-      aws-sdk-core (~> 3, >= 3.71.0)
+    aws-sdk-route53 (1.44.0)
+      aws-sdk-core (~> 3, >= 3.109.0)
       aws-sigv4 (~> 1.1)
-    aws-sdk-route53domains (1.22.0)
-      aws-sdk-core (~> 3, >= 3.71.0)
+    aws-sdk-route53domains (1.28.0)
+      aws-sdk-core (~> 3, >= 3.109.0)
       aws-sigv4 (~> 1.1)
-    aws-sdk-route53resolver (1.14.0)
-      aws-sdk-core (~> 3, >= 3.71.0)
+    aws-sdk-route53resolver (1.21.0)
+      aws-sdk-core (~> 3, >= 3.109.0)
       aws-sigv4 (~> 1.1)
-    aws-sdk-s3 (1.67.1)
-      aws-sdk-core (~> 3, >= 3.96.1)
+    aws-sdk-s3 (1.83.1)
+      aws-sdk-core (~> 3, >= 3.109.0)
       aws-sdk-kms (~> 1)
       aws-sigv4 (~> 1.1)
-    aws-sdk-securityhub (1.26.0)
-      aws-sdk-core (~> 3, >= 3.71.0)
+    aws-sdk-securityhub (1.35.0)
+      aws-sdk-core (~> 3, >= 3.109.0)
       aws-sigv4 (~> 1.1)
-    aws-sdk-ses (1.30.0)
-      aws-sdk-core (~> 3, >= 3.71.0)
+    aws-sdk-ses (1.36.0)
+      aws-sdk-core (~> 3, >= 3.109.0)
       aws-sigv4 (~> 1.1)
-    aws-sdk-sms (1.20.0)
-      aws-sdk-core (~> 3, >= 3.71.0)
+    aws-sdk-sms (1.27.0)
+      aws-sdk-core (~> 3, >= 3.109.0)
       aws-sigv4 (~> 1.1)
-    aws-sdk-sns (1.24.0)
-      aws-sdk-core (~> 3, >= 3.71.0)
+    aws-sdk-sns (1.33.0)
+      aws-sdk-core (~> 3, >= 3.109.0)
       aws-sigv4 (~> 1.1)
-    aws-sdk-sqs (1.26.0)
-      aws-sdk-core (~> 3, >= 3.71.0)
+    aws-sdk-sqs (1.34.0)
+      aws-sdk-core (~> 3, >= 3.109.0)
       aws-sigv4 (~> 1.1)
-    aws-sdk-ssm (1.80.0)
-      aws-sdk-core (~> 3, >= 3.71.0)
+    aws-sdk-ssm (1.95.0)
+      aws-sdk-core (~> 3, >= 3.109.0)
       aws-sigv4 (~> 1.1)
-    aws-sigv4 (1.1.4)
-      aws-eventstream (~> 1.0, >= 1.0.2)
+    aws-sigv4 (1.2.2)
+      aws-eventstream (~> 1, >= 1.0.2)
     azure_graph_rbac (0.17.2)
       ms_rest_azure (~> 0.12.0)
     azure_mgmt_key_vault (0.17.6)
       ms_rest_azure (~> 0.12.0)
-    azure_mgmt_resources (0.17.9)
+    azure_mgmt_resources (0.18.0)
       ms_rest_azure (~> 0.12.0)
-    azure_mgmt_security (0.18.1)
+    azure_mgmt_security (0.18.2)
       ms_rest_azure (~> 0.12.0)
-    azure_mgmt_storage (0.21.1)
+    azure_mgmt_storage (0.22.0)
       ms_rest_azure (~> 0.12.0)
     bcrypt_pbkdf (1.0.1)
     builder (3.2.4)
-    chef-config (16.1.16)
+    chef-config (16.6.14)
       addressable
-      chef-utils (= 16.1.16)
+      chef-utils (= 16.6.14)
       fuzzyurl
       mixlib-config (>= 2.2.12, < 4.0)
       mixlib-shellout (>= 2.0, < 4.0)
       tomlrb (~> 1.2)
-    chef-telemetry (1.0.8)
+    chef-telemetry (1.0.14)
       chef-config
       concurrent-ruby (~> 1.0)
       ffi-yajl (~> 2.2)
-    chef-utils (16.1.16)
+    chef-utils (16.6.14)
     coderay (1.1.3)
-    concurrent-ruby (1.1.6)
-    declarative (0.0.10)
+    concurrent-ruby (1.1.7)
+    declarative (0.0.20)
     declarative-option (0.1.0)
-    diff-lcs (1.3)
-    docker-api (1.34.2)
+    diff-lcs (1.4.4)
+    docker-api (2.0.0)
       excon (>= 0.47.0)
       multi_json
@@ -194,19 +208,18 @@ GEM
       regexp_parser (~> 1.2)
     ed25519 (1.2.4)
-    equatable (0.6.1)
     erubi (1.9.0)
-    excon (0.73.0)
-    faraday (0.17.3)
+    excon (0.78.0)
+    faraday (1.0.1)
       multipart-post (>= 1.2, < 3)
-    faraday-cookie_jar (0.0.6)
-      faraday (>= 0.7.4)
+    faraday-cookie_jar (0.0.7)
+      faraday (>= 0.8.0)
       http-cookie (~> 1.0.0)
-    faraday_middleware (0.12.2)
-      faraday (>= 0.7.4, < 1.0)
-    ffi (1.12.2)
-    ffi-yajl (2.3.3)
+    faraday_middleware (1.0.0)
+      faraday (~> 1.0)
+    ffi (1.13.1)
+    ffi-yajl (2.3.4)
       libyajl2 (~> 1.2)
     fuzzyurl (0.9.0)
-    google-api-client (0.34.1)
+    google-api-client (0.44.0)
       addressable (~> 2.5, >= 2.5.1)
       googleauth (~> 0.9)
@@ -216,11 +229,11 @@ GEM
       retriable (>= 2.0, < 4.0)
       signet (~> 0.12)
-    googleauth (0.10.0)
-      faraday (~> 0.12)
+    googleauth (0.13.0)
+      faraday (>= 0.17.3, < 2.0)
       jwt (>= 1.4, < 3.0)
       memoist (~> 0.16)
       multi_json (~> 1.11)
       os (>= 0.9, < 2.0)
-      signet (~> 0.12)
+      signet (~> 0.14)
     gssapi (1.3.0)
       ffi (>= 1.0.1)
@@ -229,26 +242,24 @@ GEM
     hana (1.3.6)
     hashie (3.6.0)
-    htmlentities (4.3.4)
     http-cookie (1.0.3)
       domain_name (~> 0.5)
     httpclient (2.8.3)
-    i18n (1.8.3)
+    i18n (1.8.5)
       concurrent-ruby (~> 1.0)
     inifile (3.0.0)
-    inspec (4.19.2)
-      faraday_middleware (~> 0.12.2)
-      inspec-core (= 4.19.2)
+    inspec (4.23.11)
+      faraday_middleware (>= 0.12.2, < 1.1)
+      inspec-core (= 4.23.11)
       train (~> 3.0)
       train-aws (~> 0.1)
       train-habitat (~> 0.1)
       train-winrm (~> 0.2)
-    inspec-core (4.19.2)
+    inspec-core (4.23.11)
       addressable (~> 2.4)
       chef-telemetry (~> 1.0)
-      faraday (>= 0.9.0)
+      faraday (>= 0.9.0, < 1.1)
       hashie (~> 3.4)
-      htmlentities (~> 4.3)
-      json_schemer (~> 0.2.1)
-      license-acceptance (>= 0.2.13, < 2.0)
+      json_schemer (>= 0.2.1, < 0.2.12)
+      license-acceptance (>= 0.2.13, < 3.0)
       method_source (>= 0.8, < 2.0)
       mixlib-log (~> 3.0)
@@ -262,12 +273,11 @@ GEM
       semverse (~> 3.0)
       sslshake (~> 1.2)
-      term-ansicolor (~> 1.7)
       thor (>= 0.20, < 2.0)
-      tomlrb (~> 1.2)
+      tomlrb (~> 1.2.0)
       train-core (~> 3.0)
       tty-prompt (~> 0.17)
       tty-table (~> 0.10)
     jmespath (1.4.0)
-    json (2.3.0)
+    json (2.3.1)
     json_schemer (0.2.11)
       ecma-re-validator (~> 0.2)
@@ -275,38 +285,37 @@ GEM
       regexp_parser (~> 1.5)
       uri_template (~> 0.7)
-    jwt (2.2.1)
-    kitchen-docker (2.10.0)
-      test-kitchen (>= 1.0.0)
-    kitchen-inspec (2.0.0)
+    jwt (2.2.2)
+    kitchen-inspec (2.2.1)
       hashie (~> 3.4)
       inspec (>= 2.2.64, < 5.0)
-      test-kitchen (>= 1.6, < 3)
+      test-kitchen (>= 2.7, < 3)
     kitchen-salt (0.6.3)
       hashie (>= 3.5)
       test-kitchen (>= 1.4)
-    kitchen-vagrant (1.6.1)
+    kitchen-vagrant (1.7.0)
       test-kitchen (>= 1.4, < 3)
     libyajl2 (1.2.0)
-    license-acceptance (1.0.19)
+    license-acceptance (2.1.2)
       pastel (~> 0.7)
       tomlrb (~> 1.2)
-      tty-box (~> 0.3)
-      tty-prompt (~> 0.18)
+      tty-box (~> 0.6)
+      tty-prompt (~> 0.20)
     little-plugger (1.1.4)
-    logging (2.2.2)
+    logging (2.3.0)
       little-plugger (~> 1.1)
-      multi_json (~> 1.10)
+      multi_json (~> 1.14)
     memoist (0.16.2)
     method_source (1.0.0)
     mini_mime (1.0.2)
-    minitest (5.14.1)
-    mixlib-config (3.0.6)
+    minitest (5.14.2)
+    mixlib-config (3.0.9)
       tomlrb
-    mixlib-install (3.12.1)
+    mixlib-install (3.12.3)
       mixlib-shellout
       mixlib-versioning
       thor
-    mixlib-log (3.0.8)
-    mixlib-shellout (3.0.9)
+    mixlib-log (3.0.9)
+    mixlib-shellout (3.1.6)
+      chef-utils
     mixlib-versioning (1.2.12)
     ms_rest (0.7.6)
@@ -319,24 +328,22 @@ GEM
       faraday-cookie_jar (~> 0.0.6)
       ms_rest (~> 0.7.6)
-    multi_json (1.14.1)
+    multi_json (1.15.0)
     multipart-post (2.1.1)
-    necromancer (0.5.1)
     net-scp (3.0.0)
       net-ssh (>= 2.6.5, < 7.0.0)
-    net-ssh (5.2.0)
+    net-ssh (6.1.0)
     net-ssh-gateway (2.0.0)
       net-ssh (>= 4.0.0)
     nori (2.6.0)
-    os (1.1.0)
-    parallel (1.19.1)
+    os (1.1.1)
+    parallel (1.19.2)
     parslet (1.8.2)
-    pastel (0.7.4)
-      equatable (~> 0.6)
+    pastel (0.8.0)
       tty-color (~> 0.5)
     pry (0.13.1)
       coderay (~> 1.1)
       method_source (~> 1.0)
-    public_suffix (4.0.5)
-    regexp_parser (1.7.0)
+    public_suffix (4.0.6)
+    regexp_parser (1.8.2)
     representable (3.0.4)
       declarative (< 0.1.0)
@@ -348,5 +355,5 @@ GEM
       rspec-expectations (~> 3.9.0)
       rspec-mocks (~> 3.9.0)
-    rspec-core (3.9.2)
+    rspec-core (3.9.3)
       rspec-support (~> 3.9.3)
     rspec-expectations (3.9.2)
@@ -370,17 +377,14 @@ GEM
       jwt (>= 1.5, < 3.0)
       multi_json (~> 1.10)
-    sslshake (1.3.0)
-    strings (0.1.8)
-      strings-ansi (~> 0.1)
+    sslshake (1.3.1)
+    strings (0.2.0)
+      strings-ansi (~> 0.2)
       unicode-display_width (~> 1.5)
       unicode_utils (~> 1.4)
     strings-ansi (0.2.0)
-    sync (0.5.0)
-    term-ansicolor (1.7.1)
-      tins (~> 1.0)
-    test-kitchen (2.5.1)
+    test-kitchen (2.7.2)
       bcrypt_pbkdf (~> 1.0)
       ed25519 (~> 1.2)
-      license-acceptance (~> 1.0, >= 1.0.11)
+      license-acceptance (>= 1.0.11, < 3.0)
       mixlib-install (~> 3.6)
       mixlib-shellout (>= 1.2, < 4.0)
@@ -388,15 +392,13 @@ GEM
       net-ssh (>= 2.9, < 7.0)
       net-ssh-gateway (>= 1.2, < 3.0)
-      thor (~> 0.19)
+      thor (>= 0.19, < 2.0)
       winrm (~> 2.0)
       winrm-elevated (~> 1.0)
       winrm-fs (~> 1.1)
-    thor (0.20.3)
+    thor (1.0.1)
     thread_safe (0.3.6)
     timeliness (0.3.10)
-    tins (1.25.0)
-      sync
-    tomlrb (1.3.0)
-    train (3.2.37)
+    tomlrb (1.2.9)
+    train (3.3.27)
       activesupport (>= 5.2.4.3, < 6.0.0)
       azure_graph_rbac (~> 0.16)
@@ -405,11 +407,11 @@ GEM
       azure_mgmt_security (~> 0.18)
       azure_mgmt_storage (~> 0.18)
-      docker-api (~> 1.26)
-      google-api-client (>= 0.23.9, < 0.35.0)
-      googleauth (>= 0.6.6, < 0.11.0)
+      docker-api (>= 1.26, < 3.0)
+      google-api-client (>= 0.23.9, < 0.44.1)
+      googleauth (>= 0.6.6, < 0.13.1)
       inifile (~> 3.0)
-      train-core (= 3.2.37)
+      train-core (= 3.3.27)
       train-winrm (~> 0.2)
-    train-aws (0.1.16)
+    train-aws (0.1.18)
       aws-sdk-apigateway (~> 1.0)
       aws-sdk-apigatewayv2 (~> 1.0)
@@ -418,4 +420,5 @@ GEM
       aws-sdk-budgets (~> 1.0)
       aws-sdk-cloudformation (~> 1.0)
+      aws-sdk-cloudfront (~> 1.0)
       aws-sdk-cloudhsm (~> 1.0)
       aws-sdk-cloudhsmv2 (~> 1.0)
@@ -441,4 +444,5 @@ GEM
       aws-sdk-elasticsearchservice (~> 1.0)
       aws-sdk-firehose (~> 1.0)
+      aws-sdk-guardduty (~> 1.31)
       aws-sdk-iam (~> 1.13)
       aws-sdk-kafka (~> 1.0)
@@ -459,36 +463,34 @@ GEM
       aws-sdk-sqs (~> 1.10)
       aws-sdk-ssm (~> 1.0)
-    train-core (3.2.37)
+    train-core (3.3.27)
       addressable (~> 2.5)
-      ffi (< 1.13)
+      ffi (!= 1.13.0)
       json (>= 1.8, < 3.0)
       mixlib-shellout (>= 2.0, < 4.0)
       net-scp (>= 1.2, < 4.0)
-      net-ssh (>= 2.9, < 6.0)
+      net-ssh (>= 2.9, < 7.0)
     train-habitat (0.2.13)
-    train-winrm (0.2.6)
+    train-winrm (0.2.11)
       winrm (~> 2.0)
+      winrm-elevated (~> 1.2.2)
       winrm-fs (~> 1.0)
-    tty-box (0.5.0)
-      pastel (~> 0.7.2)
-      strings (~> 0.1.6)
+    tty-box (0.6.0)
+      pastel (~> 0.8)
+      strings (~> 0.2.0)
       tty-cursor (~> 0.7)
-    tty-color (0.5.1)
+    tty-color (0.5.2)
     tty-cursor (0.7.1)
-    tty-prompt (0.21.0)
-      necromancer (~> 0.5.0)
-      pastel (~> 0.7.0)
-      tty-reader (~> 0.7.0)
-    tty-reader (0.7.0)
+    tty-prompt (0.22.0)
+      pastel (~> 0.8)
+      tty-reader (~> 0.8)
+    tty-reader (0.8.0)
       tty-cursor (~> 0.7)
-      tty-screen (~> 0.7)
-      wisper (~> 2.0.0)
-    tty-screen (0.8.0)
-    tty-table (0.11.0)
-      equatable (~> 0.6)
-      necromancer (~> 0.5)
-      pastel (~> 0.7.2)
-      strings (~> 0.1.5)
-      tty-screen (~> 0.7)
+      tty-screen (~> 0.8)
+      wisper (~> 2.0)
+    tty-screen (0.8.1)
+    tty-table (0.12.0)
+      pastel (~> 0.8)
+      strings (~> 0.2.0)
+      tty-screen (~> 0.8)
     tzinfo (1.2.7)
       thread_safe (~> 0.1)
@@ -500,5 +502,5 @@ GEM
     unicode_utils (1.4.0)
     uri_template (0.7.0)
-    winrm (2.3.4)
+    winrm (2.3.5)
       builder (>= 2.1.2)
       erubi (~> 1.8)
@@ -509,5 +511,5 @@ GEM
       nori (~> 2.0)
       rubyntlm (~> 0.6.0, >= 0.6.1)
-    winrm-elevated (1.2.1)
+    winrm-elevated (1.2.2)
       erubi (~> 1.8)
       winrm (~> 2.0)
@@ -524,7 +526,7 @@ PLATFORMS
 
 DEPENDENCIES
-  kitchen-docker (>= 2.9)
-  kitchen-inspec (>= 1.1)
-  kitchen-salt (>= 0.6.0)
+  kitchen-docker!
+  kitchen-inspec (>= 2.2.1)
+  kitchen-salt (>= 0.6.3)
   kitchen-vagrant
   rspec-retry
```

Even if we do figure out which gem(s) it is, we can't really pin to an older version.  So the real question is what to do about the original question of text-based vs. YAML-based comparisons.

Reply:
As @baby-gnu reported in Slack/IRC, the regression was introduced by this PR:

* https://github.com/inspec/inspec/pull/5054

That PR was included in this release:

* https://github.com/inspec/inspec/releases/tag/v4.23.4

Confirmed that locally by testing the gems and the last working version is the one before:

* https://rubygems.org/gems/inspec/versions/4.22.22

Furthermore, this version is actually _working_ for YAML-based comparisons, so that avoids the whole issue of modifying the output from Windows.  I'll adjust this PR to do that as well.

Reply:
Original point is hardly an issue now due to YAML-based comparison and the gem issue has been resolved (pushed already to all formulas using the `ssf-formula`).